test/utils.cpp

```diff
@@ -373,6 +373,21 @@ TEST_CASE("extract_filter()", "[utils]")
 
 	REQUIRE(filter == "~/bin/script.sh");
 	REQUIRE(url == "https://newsboat.org");
+
+	utils::extract_filter("https://newsboat.org", filter, url);
+
+	REQUIRE(filter == "//newsboat.org");
+	REQUIRE(url == "https://newsboat.org");
```

Comment:
The current behavior is quite "interesting" for some edge cases :slightly_smiling_face:. It doesn't really matter, since it is only called on strings which satisfy `is_filter`.
The question is: Do we want to match the current behavior or if it is OK if the Rust version behaves differently?

Reply:
I think it's enough to match the behaviour for cases where we know that input starts with "filter:". In other words, assume `utils::is_filter_url(input) == true`.

Reply:
This is invalid.